tests/test_settings.py

```diff
@@ -0,0 +1,265 @@
+
+
+def test_change_display_name(client):
+    # Register and log in a user
+    user = register_user(client, "testuser_settings", "SecureTestPass123!")
+    assert user is not None, "User registration failed"
+
+    login_user(client, "testuser_settings", "SecureTestPass123!")
+
+    # Define new display name
+    new_display_name = "New Display Name"
+
+    # Submit POST request to change display name
+    response = client.post(
+        "/settings/",
+        data={
+            "display_name": new_display_name,
+            "update_display_name": "Update Display Name",
+        },
+        follow_redirects=True,
+    )
+
+    # Verify update was successful
+    assert response.status_code == 200, "Failed to update display name"
+
+    # Fetch updated user info from the database to confirm change
+    updated_user = User.query.filter_by(primary_username="testuser_settings").first()
+    assert updated_user.display_name == new_display_name, "Display name was not updated correctly"
+
+    # Optional: Check for success message in response
+    assert (
+        b"Display name updated successfully" in response.data
+    ), "Success message not found in response"
+
+
+def test_change_username(client):
+    # Register and log in a user
+    user = register_user(client, "original_username", "SecureTestPass123!")
+    assert user is not None, "User registration failed"
+
+    login_user(client, "original_username", "SecureTestPass123!")
+
+    # Define new username
+    new_username = "updated_username"
+
+    # Submit POST request to change the username
+    response = client.post(
+        "/settings/",
+        data={
+            "new_username": new_username,
+            "change_username": "Update Username",  # This button name must match your HTML form
+        },
+        follow_redirects=True,
+    )
+
+    # Verify update was successful
+    assert response.status_code == 200, "Failed to update username"
+
+    # Fetch updated user info from the database to confirm change
+    updated_user = User.query.filter_by(primary_username=new_username).first()
+    assert updated_user is not None, "Username was not updated correctly in the database"
+    assert (
+        updated_user.primary_username == new_username
+    ), "Database does not reflect the new username"
+
+    assert (
+        not updated_user.is_verified
+    ), "User verification status should be reset after username change"
+
+    # Optional: Check for success message in response
+    assert (
+        b"Username changed successfully" in response.data
+    ), "Success message not found in response"
+
+
+def test_add_pgp_key(client):
+    # Setup and login
+    user = register_user(client, "user_with_pgp", "SecureTestPass123!")
+    assert user is not None, "User registration failed"
+    login_user(client, "user_with_pgp", "SecureTestPass123!")
+
+    # Define a PGP key string (use a realistic dummy key for actual testing)
+    new_pgp_key = """-----BEGIN PGP PUBLIC KEY BLOCK-----
```

Comment:
move this to a standalone file in the `tests` dir and just load it. makes it easier to skim the code

Reply:
👍